crates/in_stream/src/ws.rs

```diff
@@ -270,6 +270,16 @@ impl<Sub: InStreamStd> InStream<&mut WsFrame, WsFrame> for InStreamWss<Sub> {
         }
     }
 
+    fn remote_url(&self) -> Url2 {
+        let mut url = match self.state.as_ref().unwrap() {
+            WssState::MidCliHandshake(s) => s.get_ref().get_ref().remote_url(),
+            WssState::MidSrvHandshake(s) => s.get_ref().get_ref().remote_url(),
+            WssState::Ready(s) => s.get_ref().remote_url(),
```

Comment:
if they are the same type we can probably get away with a 
WssState::MidCliHandshake(s) | WssState::MidSrvHandshake(s) | WssState::Ready(s) => s.get_ref().get_ref().remote_url(),

Reply:
alas, they are not, tungstenite has different structs for each of these cases.